identity/src/main/java/com/stripe/android/identity/states/LaplacianBlurDetector.kt

```diff
@@ -0,0 +1,158 @@
+            }
+            blurTargetAllocation.copyTo(smootherBitmap)
+
+            // Greyscale so we're only dealing with white <--> black pixels, this is so we only need to
+            // detect pixel luminosity
+            val greyscaleBitmap = Bitmap.createBitmap(
+                sourceBitmap.width,
+                sourceBitmap.height,
+                sourceBitmap.config
+            )
+            val smootherInput = Allocation.createFromBitmap(
+                renderScript,
+                smootherBitmap,
+                Allocation.MipmapControl.MIPMAP_NONE,
+                Allocation.USAGE_SHARED
+            )
+            val greyscaleTargetAllocation = Allocation.createFromBitmap(
+                renderScript,
+                greyscaleBitmap,
+                Allocation.MipmapControl.MIPMAP_NONE,
+                Allocation.USAGE_SHARED
+            )
+
+            // Inverts and greyscales the image
+            val colorIntrinsic = ScriptIntrinsicColorMatrix.create(renderScript)
+            colorIntrinsic.setGreyscale()
+            colorIntrinsic.forEach(smootherInput, greyscaleTargetAllocation)
+            greyscaleTargetAllocation.copyTo(greyscaleBitmap)
+
+            // Run edge detection algorithm using a laplacian matrix convolution
+            // Apply 3x3 convolution to detect edges
+            val edgesBitmap = Bitmap.createBitmap(
+                sourceBitmap.width,
+                sourceBitmap.height,
+                sourceBitmap.config
+            )
+            val greyscaleInput = Allocation.createFromBitmap(
+                renderScript,
+                greyscaleBitmap,
+                Allocation.MipmapControl.MIPMAP_NONE,
+                Allocation.USAGE_SHARED
+            )
+            val edgesTargetAllocation = Allocation.createFromBitmap(
+                renderScript,
+                edgesBitmap,
+                Allocation.MipmapControl.MIPMAP_NONE,
+                Allocation.USAGE_SHARED
+            )
+
+            val convolve =
+                ScriptIntrinsicConvolve3x3.create(renderScript, Element.U8_4(renderScript))
+            convolve.setInput(greyscaleInput)
+            convolve.setCoefficients(CLASSIC_MATRIX) // Or use others
+            convolve.forEach(edgesTargetAllocation)
+            edgesTargetAllocation.copyTo(edgesBitmap)
+
+            // Scale the score from 0~255(0xFF)
+            return mostLuminousIntensityFromBitmap(edgesBitmap).toFloat() / COLOR_MAX
+        } catch (e: Exception) {
+            Log.e(TAG, "Failed to calculate blur score. $e")
+            return DEFAULT_SCORE
+        }
+    }
+
+    /**
+     * Resolves the most luminous color pixel in a given greyscale bitmap.
+     *
+     * @param bitmap Source greyscale bitmap, with same RGB channel value.
+     * @return The most luminous color in bitmap, ranging from 0x00 to 0xFF.
+     */
+    private fun mostLuminousIntensityFromBitmap(bitmap: Bitmap): Int {
```

Comment:
Note: this is different from https://www.zacsweers.dev/detecting-blurriness-with-renderscript/, at this point the bitmap is already grayscale with equal values in RGB, so we only need to get any channel's 8bit value as an indicator of luminous

Reply:
nit: might make sense to name this `mostLuminousIntensityFromGreyscaleBitmap` just to clarify.